BUILD.gn

```diff
@@ -0,0 +1,280 @@
+# Copyright (c) 2013-2019 GitHub Inc.
+# Copyright 2019 the V8 project authors. All rights reserved.
+# Copyright 2023 Microsoft Inc.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This file is used by GN for building, which is NOT the build system used for
+# building official binaries.
+# Please take a look at node.gyp if you are making changes to build system.
```

Comment:
The comment here doesn't seem striking enough to ensure that people would not accidentally try to edit these files...is it possible to move the actual configs into some file with a more obvious name e.g. `unofficial.gn` or `unsupported.gn`? Or leaving the bulk of the configuration in `unofficial.gni` or `unsupported.gni` and leaving the `BUILD.gn` basically just that comment and some very thin wrappers around the imported configs?

Reply:
GN requires the config file name to be `BUILD.gn` so I can not rename it to `unofficial.gn`.

I have also tried to split the file but GN explicitly forbids putting actual configuration in gni file:

```
ERROR at //node/unofficial.gni:18:1: Not valid from an import.
config("node_features") {
^------------------------
Imports are for defining defaults, variables, and rules. The
appropriate place for this kind of thing is really in a normal
BUILD file.
```

I also tried to move the `BUILD.gn` from root dir to `src/BUILD.gn`, but this approach doesn't work well. The GN file reads variables from `node.gyp`, where filenames are relative to the root dir instead of `src` dir, so to move `BUILD.gn` to `src/BUILD.gn` I have to add script to change filenames on the fly, which would make the GN files more fragile to changes in gyp world.

Maybe I can put a large warning like below in `BUILD.gn`?

```
####################################################
#                                                  #
#            DO NOT EDIT THIS FILE!                #
#                                                  #
####################################################
```

Reply:
> I have also tried to split the file but GN explicitly forbids putting actual configuration in gni file:

From what I can tell from the chromium configs, it seems the convention is to put these in a template (there are quite a lot of these examples actually, I just searched for `config(` in `**/*.gni`).

Reply:
Thanks for the hint, it works quite well!

I have updated the code to separate the gn file.